controllers/module_reconciler.go

```diff
@@ -340,21 +345,18 @@ func (mrh *moduleReconcilerHelper) handleSigning(ctx context.Context, mld *api.M
 }
 
 func (mrh *moduleReconcilerHelper) handleDriverContainer(ctx context.Context,
-	mld *api.ModuleLoaderData,
-	dsByKernelVersion map[string]*appsv1.DaemonSet) error {
+	mld *api.ModuleLoaderData) error {
+
+	hashValue, err := hashstructure.Hash(hashData{kernelVersion: mld.KernelVersion}, nil)
+	if err != nil {
+		return fmt.Errorf("failed to hash kernel version %s for module loader daemonset name: %v", mld.KernelVersion, err)
+	}
+	name := mld.Name + "-" + fmt.Sprintf("%d", hashValue)
```

Comment:
How about we use hex to same some characters? Also, let's do all the formatting in `fmt.Sprintf`:
```suggestion
	name := fmt.Sprintf("%s-%x", mld.Name, hashValue)
```

Reply:
fixed